### PR TITLE
Make target CPU and container resources configurable

### PR DESCRIPTION
**Pull request by jackkleeman**
We had fine defaults, but some containers are going to burn more CPU, eg
if they have busy long lived connections, and also we should make the
HPA target cpu configurable, and bring down the default a little. Its an
average, after all.

### CODE REVIEW

File: controllers/deployment.go [hunk 1]
```diff
@@ -64,6 +64,22 @@ func deployment(es *egressv1.ExternalService, configHash string) *appsv1.Deploym
...
+	if es.Spec.Resources != nil {
+		resources = *es.Spec.Resources
+	} else {
+		resources = corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				"cpu":    resource.MustParse("100m"),
+				"memory": resource.MustParse("50Mi"),
+			},
+			Limits: corev1.ResourceList{
+				"cpu":    resource.MustParse("2"),
```
Comment (chongyangsh) [inline controllers/deployment.go]: Is this default limit a bit high? I guess it doesn't matter as resource allocation is only guaranteed up to requests.

File: api/v1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -123,6 +123,16 @@ func (in *ExternalServiceSpec) DeepCopyInto(out *ExternalServiceSpec) {
 		*out = new(int32)
 		**out = **in
 	}
+	if in.TargetCPUUtilizationPercentage != nil {
+		in, out := &in.TargetCPUUtilizationPercentage, &out.TargetCPUUtilizationPercentage
+		*out = new(int32)
+		**out = **in
```
Comment (chongyangsh) [inline api/v1/zz_generated.deepcopy.go]: OOI what does this do?

Comment (jackkleeman) [reply api/v1/zz_generated.deepcopy.go]: This is autogenerated code allowing deep copying of the object. I guess its a double dereference setting a double pointer?

File: controllers/deployment.go (hunk 1 above)
Comment (jackkleeman) [reply controllers/deployment.go]: I don't think it can hurt

**Pull request merged**